infra/vsts_agent_linux_startup.sh

```diff
@@ -95,6 +96,94 @@ useradd \
 #add docker group to user
 usermod -aG docker vsts
 
+# The Bazel cache fills up the hard drive in less than a day, and is a real
+# pain to clean up, as it consists of a very large amount of read-only files in
+# a fairly large tree of read-only directories. Therefore, it looks like the
+# best way to manage it is to mount it on a partition and reset the partition.
+
+cat <<REMOUNT_CACHE > /root/remount_cache.c
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+int exists(const char* path) {
+	struct stat s;
+	return stat(path, &s) == 0 && S_ISDIR(s.st_mode);
+}
+
+int execute_f(char** args) {
+	int status;
+	char** child_env = { NULL };
+	pid_t pid = fork();
+	if (0 == pid) {
+		// child process
+		if (-1 == execve(args[0], args, child_env)) {
```

Comment:
relying on a macro to enforce that this is not an out of bound scares me a bit. I guess varargs aren’t that much safer but at least more transparent than cpp.

Reply:
Looking through the documentation on C varargs, I have not found any way to get at the number of arguments passed. Like so many other decisions in this PR, I can assure you going with a cpp macro was not my first choice, but it did seem to be the least-worst available.

Actually... All I need here is an executable. It will have to run as root, but perhaps it does not need to be written in C. How would you feel about this if it were written in Haskell?

Reply:
Also just to be clear, I'm relying on the macro to ensure that the arg list is properly terminated (i.e. has a `NULL` as the last argument) without too much visual clutter in `execute` calls. I'm relying on myself to ensure bounds are not an issue. (I'm not claiming this is reassuring.)